src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -106,7 +106,11 @@ def build_expr_from_python_val(expr_builder: proto.Expr, obj: Any) -> None:
         unscaled_val = reduce(lambda val, digit: val * 10 + digit, dec_tuple.digits)
         if dec_tuple.sign != 0:
             unscaled_val *= -1
-        req_bytes = (unscaled_val.bit_length() + 7) // 8
+
+        # In two-complement -1 with one byte is 0xFF. We encode arbitrary length integers
+        # in full bytes. Therefore, round up to fullest byte. To restore the sign, add another byte.
+        req_bytes = unscaled_val.bit_length() // 8 + 1
```

Comment:
This allocates one extra byte for positive powers of 256. If allocating the optimal number of bytes is a goal, the previous formula works well.

Reply:
that's correct I believe, we need to have the extra byte to distinguish between positive and negative. Technically a bit would be sufficient, but we round up to full bytes.